### PR TITLE
Included bucket check

### PR DESCRIPTION
**Pull request by rmsmani**
**Please refer to the contribution guidelines in the README when submitting PRs.**

Included bucket check

### CODE REVIEW

File: localstack/services/s3/s3_listener.py [hunk 1]
```diff
@@ -634,7 +634,7 @@ def strip_chunk_signatures(data):
 
 
 def is_bucket_available(bucket_name):
-    body = {}
+    body = {''}
```
Comment (whummer) [inline localstack/services/s3/s3_listener.py:637]: What is the purpose of this change? Note that `body` should be of type `dict`, not `set`. With your change, line 637 would create a `set` with a single item in it..?

Comment (rmsmani) [reply localstack/services/s3/s3_listener.py:637]: Ok will fix it

**Comment by rmsmani** [comment]
@whummer 
Please review and merge

File: localstack/services/s3/s3_listener.py [hunk 2]
```diff
@@ -628,6 +633,25 @@ def strip_chunk_signatures(data):
...
+    exists, code = bucket_exists(bucket_name)
+    if not exists:
+        body = {
+            'Error': {
+                'Code': code,
+                'Message': 'The bucket does not exist'
+            }
+        }
+        exists = False
+        code = code
```
Comment (whummer) [inline localstack/services/s3/s3_listener.py:647]: This line is a no-op and can be removed

File: localstack/services/s3/s3_listener.py [hunk 3]
```diff
@@ -508,8 +508,13 @@ def remove_xml_preamble(response):
...
     bucket_name = normalize_bucket_name(bucket_name)
+    exists, code, body = is_bucket_available(bucket_name)
+    if not exists:
+        return requests_response(body, status_code=code)
+
     lifecycle = BUCKET_LIFECYCLE.get(bucket_name)
     status_code = 200
+
     if not lifecycle:
         # TODO: check if bucket actually exists
```
Comment (whummer) [inline localstack/services/s3/s3_listener.py:519]: Can we move lines 511-513 to within the `if`-branch in this line, and remove the `TODO` statement?

File: localstack/services/s3/s3_listener.py [hunk 4]
```diff
@@ -628,6 +633,25 @@ def strip_chunk_signatures(data):
...
+                'Code': code,
+                'Message': 'The bucket does not exist'
+            }
+        }
+        exists = False
+        code = code
+        xmltodict.unparse(body)
+        return exists, code, body
+
+    xmltodict.unparse(body)
```
Comment (whummer) [inline localstack/services/s3/s3_listener.py:651]: I think we can remove lines 651 and 648 - calling `xmltodict.unparse(body)` has no side-effects, and we're not using the results returned from these functions.

**Comment by rmsmani** [comment]
@whummer Fixed the review comments.
